connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -384,6 +452,40 @@ public void onCompletion(RecordMetadata recordMetadata, Exception e) {
         return true;
     }
 
+    // Due to transformations that may change the destination topic of a record (such as
+    // RegexRouter) topic creation can not be batched for multiple topics
+    private void maybeCreateTopic(String topic) {
+        if (!topicCreation.isTopicCreationEnabled() || topicCreation.topicCache().contains(topic)) {
+            return;
+        }
+        log.info("The task will send records to topic '{}' for the first time. Checking "
+                + "whether topic exists", topic);
+        Map<String, TopicDescription> existing = admin.describeTopics(topic);
+        if (!existing.isEmpty()) {
+            log.info("Topic '{}' already exists.", topic);
+            topicCreation.topicCache().add(topic);
```

Comment:
Another thing for potentially moving into `TopicCreation`, such as maybe adding a `addTopic(topic)` method. Doing so might allow you to hide most of the implementation details of `TopicCreation`.
```suggestion
            topicCreation.addTopic(topic);
```